travis.sh

```diff
@@ -67,6 +67,59 @@ build_java_oracle8() {
   build_java
 }
 
+build_android() {
+  # Set up Gradle, then update to latest version.
+  # http://www.tothenew.com/blog/gradle-installation-in-ubuntu/
+  sudo apt-get -y install python-software-properties # for apt-add-repository
+  yes | sudo add-apt-repository ppa:cwchien/gradle
+  sudo apt-get update
+  sudo apt-get -y install gradle-2.11
+  gradle -v
+  # Set up the Android SDK.
+  # https://gist.github.com/wenzhixin/43cf3ce909c24948c6e7
+  # We can't use the default Android Travis runtime, because we are a
+  # multi-language project.
+  # Need 32-bit versions of the libraries.
+  # http://stackoverflow.com/a/19524010/101923
+  sudo dpkg --add-architecture i386
+  sudo apt-get -qqy update
+  # adb
+  sudo apt-get -y install libc6:i386 libstdc++6:i386
+  # aapt
+  sudo apt-get -y install zlib1g:i386
+  wget http://dl.google.com/android/android-sdk_r24.4.1-linux.tgz
+  tar xzf android-sdk_r24.4.1-linux.tgz
+  export ANDROID_HOME=$PWD/android-sdk-linux
+  export PATH="${PATH}:${ANDROID_HOME}/tools:${ANDROID_HOME}/platform-tools"
+  # Accept the license.
+  # http://stackoverflow.com/a/17863931/101923
+  sudo apt-get -y install expect
+  expect -c '
+  set timeout -1   ;
```

Comment:
I assume `-1` means no timeout? Probably better to set it to something unreasonably high, just so we get _some_ sort of error if something goes horribly wrong..